unstructured/ingest/cli/cmds/sharepoint.py

```diff
@@ -55,6 +58,24 @@ def add_cli_options(cmd: click.Command) -> None:
                     https://[tenant]-admin.sharepoint.com.\
                     This requires the app to be registered at a tenant level",
             ),
+            click.Option(
```

Comment:
Can we actually add this via a `PermissionConfig` data class similar to how we add read, and partition configs to the list of options available on a click command? Then we can also pull the data out using `from_dict`

Reply:
Please let me know if there are additional concerns on this one

Reply:
@ahmetmeleq now that you have CliPermissionsConfig can we remove these options explicitly defined here? and instead add via add_options[ here](https://github.com/Unstructured-IO/unstructured/blob/569561e59bdcead6953f87d1e18dcbe32ada78e7/unstructured/ingest/cli/cmds/sharepoint.py#L95)?

Reply:
Oh right, I thought this is for the future for each pipeline node to have their own config. Now I realize how it immediately reduces the number of explicitly defined options in the connector cli config. I added https://github.com/Unstructured-IO/unstructured/pull/1628/commits/ae982a4e4846864ee94ae1ac15c29614fbcc8abb and https://github.com/Unstructured-IO/unstructured/pull/1628/commits/b898ed5d3fa3b90f74872906bb6bf7b9f5f84209 👍

Reply:
yep, exactly!